### PR TITLE
Adds Fine permission UI to background permission sample.

### PR DESCRIPTION
**Pull request by codingjeremy**

### CODE REVIEW

File: LocationUpdatesBackgroundKotlin/app/src/main/java/com/google/android/gms/location/sample/locationupdatesbackgroundkotlin/PermissionRequestFragment.kt [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+    private lateinit var binding: FragmentPermissionRequestBinding
+
+    private var activityListener: Callbacks? = null
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        arguments?.let {
```
Comment (nic0lette) [inline LocationUpdatesBackgroundKotlin/app/src/main/java/com/google/android/gms/location/sample/locationupdatesbackgroundkotlin/PermissionRequestFragment.kt]: Maybe consider something like: `(arguments as? PermissionRequestType)?.let {` ?

Comment (codingjeremy) [reply LocationUpdatesBackgroundKotlin/app/src/main/java/com/google/android/gms/location/sample/locationupdatesbackgroundkotlin/PermissionRequestFragment.kt]: I need to cast the returned type from arguments.getSerializable(ARG_PERMISSION_REQUEST_TYPE) as PermissionRequestType, not the instance of arguments.  I guess I could do the code below but that seems more confusing to me than how it is written now. Any other ideas?

        (arguments?.getSerializable(ARG_PERMISSION_REQUEST_TYPE) as? PermissionRequestType)?.let {
            permissionRequestType = it
        }

Comment (codingjeremy) [reply LocationUpdatesBackgroundKotlin/app/src/main/java/com/google/android/gms/location/sample/locationupdatesbackgroundkotlin/PermissionRequestFragment.kt]: or maybe...

        permissionRequestType = arguments?.getSerializable(ARG_PERMISSION_REQUEST_TYPE) as PermissionRequestType

Comment (nic0lette) [reply LocationUpdatesBackgroundKotlin/app/src/main/java/com/google/android/gms/location/sample/locationupdatesbackgroundkotlin/PermissionRequestFragment.kt]: x.x I was apparently too tired when I read through this. Yeah, the way you wrote it seems easiest to follow?
So yeah, let's just call it good.

Comment (codingjeremy) [reply LocationUpdatesBackgroundKotlin/app/src/main/java/com/google/android/gms/location/sample/locationupdatesbackgroundkotlin/PermissionRequestFragment.kt]: Ha ha :D 

Sounds good, changed.

**Pull request merged**